src/main/domain/vector-search.ts

```diff
@@ -0,0 +1,569 @@
+import sqlite3 from "sqlite3";
+import path from "path";
+import fs from "fs-extra";
+import log from "electron-log/main";
+import { app } from "electron";
+import { RecursiveCharacterTextSplitter } from "langchain/text_splitter";
+import { Document } from "@langchain/core/documents";
+import * as history from "./history";
+import { getEmbeddings } from "./llm";
+import { RecordingTranscript, RecordingTranscriptItem } from "../../types";
+import { getSettings } from "./settings";
+
+interface VectorSearchResult {
+  recordingId: string;
+  chunkIndex: number;
+  text: string;
+  timestamp: string;
+  speaker: string;
+  score: number;
+  startTime: number;
+  endTime: number;
+}
+
+interface TranscriptChunk {
+  id: string;
+  recordingId: string;
+  chunkIndex: number;
+  text: string;
+  timestamp: string;
+  speaker: string;
+  startTime: number;
+  endTime: number;
+  embedding: number[];
+}
+
+class SQLiteVectorStore {
+  private db: sqlite3.Database | null = null;
+  private embeddings: any = null;
+  private dbPath: string;
+
+  constructor() {
+    // Use user data directory for packaged app, fallback to cwd for development
+    const dataDir = app.isPackaged 
+      ? app.getPath('userData')
+      : process.cwd();
+    this.dbPath = path.join(dataDir, "vector-store", "vector-store.db");
+  }
+
+  async initialize() {
+    try {
+      // Ensure directory exists
+      await fs.ensureDir(path.dirname(this.dbPath));
+
+      // Initialize the embedding model
+      log.info("Loading embedding model...");
+      this.embeddings = await getEmbeddings();
+      log.info("Embedding model loaded successfully");
+
+      // Open SQLite database
+      log.info("Opening SQLite database...");
+      this.db = new sqlite3.Database(this.dbPath);
+      
+      // Load sqlite-vec extension if available
+      await this.loadSqliteVec();
+
+      // Create tables
+      await this.createTables();
+      
+      log.info("SQLite vector store initialized successfully");
+    } catch (error) {
+      log.error("Failed to initialize SQLite vector store:", error);
+      throw error;
+    }
+  }
+
+  private async loadSqliteVec() {
+    if (!this.db) return;
+
+    try {
+      // Try to load sqlite-vec extension
+      // Note: This requires the sqlite-vec extension to be available
+      // For now, we'll use basic SQLite with FTS5 for text search
+      log.info("sqlite-vec extension not available, using FTS5 for text search");
+    } catch (error) {
+      log.warn("Could not load sqlite-vec extension:", error);
+    }
+  }
+
+  private async createTables() {
+    if (!this.db) return;
+
+    return new Promise<void>((resolve, reject) => {
+      this.db!.serialize(() => {
```

Comment:
Please use checks instead of non-null-assertions, like calling `if (!this.db) return;` within the promise.